src/authenticatedFetch/dpop/DpopAuthenticatedFetcher.ts

```diff
@@ -31,16 +31,84 @@ import { IFetcher } from "../../util/Fetcher";
 import { IDpopHeaderCreator } from "../../dpop/DpopHeaderCreator";
 import { IUrlRepresentationConverter } from "../../util/UrlRepresenationConverter";
 import { IStorageUtility } from "../../localStorage/StorageUtility";
-import { Headers as NodeHeaders } from "cross-fetch";
 
+/**
+ * @internal
+ * This method is a temporary fix: as per the MDN spec (https://developer.mozilla.org/en-US/docs/Web/API/Headers),
+ * the Headers object should have a `.keys()` method. It turns out that some implementations
+ * do not implement this method, so this check is necessary to verify how to access
+ * the header's content.
+ * @param headers A header potentially of the type provided by LDflex-Comunica
+ */
+function hasKeys(
+  headers: Headers | string[][] | Record<string, string> | undefined | unknown
+): headers is Headers | string[][] {
+  return (
+    // eslint-disable-next-line
+    // @ts-ignore
+    headers !== undefined && headers.keys !== undefined
+  );
+}
+
+/**
+ * @internal
+ * This method is a temporary fix: we receive from LDflex-Comunica an object
+ * that is not conform to the Headers interface. This mitigates the issue
+ * until a cleaner fix is found. This method may be seen as a type guard for
+ * a LDflex-Comunica header.
+ * @param headers A header potentially of the type provided by LDflex-Comunica
+ */
+function hasRawGetter(
+  // The 'any' type here is needed because no type definition covers properly
+  // the headers object we want to check against here.
+  // eslint-disable-next-line
+  headers: any
+): boolean {
+  return (
+    // eslint-disable-next-line
+    // @ts-ignore
+    headers !== undefined && headers.raw !== undefined
+  );
+}
+
+/**
+ * @internal
+ * This function feels unnecessarily complicated, but is required in order to
+ * have Headers according to type definitions in both Node and browser environments.
+ * This might require a fix upstream to be cleaned up.
+ *
+ * @param headersToFlatten A structure containing headers potentially in several formats
+ */
 export function flattenHeaders(
   headersToFlatten: Headers | string[][] | Record<string, string> | undefined
 ): Record<string, string> {
+  if (headersToFlatten === undefined) {
+    return {};
+  }
```

Comment:
The functions in this file seem like they could be used in other classes as well (The the bearer authenticated fetcher). A new class should be created called `HeaderFlattener` or something like that and it should be injected into this class and any other class that makes a fetch.

Reply:
I also now notice that the `if (!authToken) {` on line 85 should not actually be in this file. Instead there should be another handler called `UnauthenticatedFetcher` that handles the case where the auth token isn't present.